File: src/TypeScript/TypeScriptLibrary/Library/Measurements/ObjectTransformerLink.ts
import { constructor } from "assert";
import { CategoryArrow } from "../CategoryArrow";
import { ICategoryObject } from "../Interfaces/ICategoryObject";
import { IDesktop } from "../Interfaces/IDesktop";
import { IDataConsumer } from "./Interfaces/IDataConsumer";
import { IMeasurements } from "./Interfaces/IMeasurements";
import { IObjectTransformer } from "./Interfaces/IObjectTransformer";
import { IObjectTransformerConsumer } from "./Interfaces/IObjectTransformerConsumer";

export class ObjectTransformerLink extends CategoryArrow {
    constructor(desktop: IDesktop, name: string) {
        super(desktop, name)
        this.typeName = "ObjectTransformerLink";
        this.types.push("ObjectTransformerLink");
    }

    getSource(): ICategoryObject {
        return this.consumer as unknown as ICategoryObject;
    }
    getTagret(): ICategoryObject {
        return this.transformer as unknown as ICategoryObject;
    }
    setSource(source: ICategoryObject): void
    {
        this.consumer = source as unknown as IObjectTransformerConsumer;
    }
    setTarget(target: ICategoryObject): void {
        this.transformer = target as unknown as IObjectTransformer;
        this.consumer.addTransformer(this.transformer);
    }


    transformer !: IObjectTransformer;
    consumer !: IObjectTransformerConsumer;


}

File: Java_Generated/Java/Trading1/app/src/main/java/measurements/arrows/ObjectTransformerLink.java
package measurements.arrows;

import category_theory.CategoryArrow;
import category_theory.interfaces.ICategoryObject;
import diagram.interfaces.IDesktop;
import measurements.interfaces.IObjectTransformer;
import measurements.interfaces.IObjectTransformerConsumer;

import java.util.Map;

public class ObjectTransformerLink extends CategoryArrow {
    public ObjectTransformerLink(String name, IDesktop desktop) {
        super(name, desktop);
    }

    /**
     * @param source
     */
    @Override
    public void setSource(ICategoryObject source) {
        super.setSource(source);
        consumer = (IObjectTransformerConsumer) source;
    }


    /**
     * @param target
     */
    @Override
    public void setTarget(ICategoryObject target) {
        super.setTarget(target);
        transformer = (IObjectTransformer) target;
        consumer.addTransfotmer(transformer);
    }

    IObjectTransformerConsumer consumer;

    IObjectTransformer transformer;
}
